### PR TITLE
feat: migrate to GAR for runner images

### LINKED ISSUE #91: Artifact Registry for storing the Terraform image
Using a GAR repo for storing the Terraform image will allow for granular permissions and reuse of the worker image in other pipelines

### PR DESCRIPTION
**Pull request by bharathkkb**
fixes #91

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./test/integration/cloudbuild_enabled/controls/gcloud.rb:19:  
./test/integration/cloudbuild_enabled/controls/gcloud.rb:37:  
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
 ```

File: modules/cloudbuild/main.tf [hunk 1]
```diff
@@ -255,9 +256,16 @@ resource "google_storage_bucket_iam_member" "cloudbuild_artifacts_iam" {
...
-  ]
+}
+
+resource "google_artifact_registry_repository_iam_member" "terraform-image-iam" {
+  provider = google-beta
+  project  = module.cloudbuild_project.project_id
+
+  location   = google_artifact_registry_repository.tf-image-repo.location
+  repository = google_artifact_registry_repository.tf-image-repo.name
+  role       = "roles/artifactregistry.writer"
```
Comment (rjerrems) [inline modules/cloudbuild/main.tf:267]: is reader provided by default?

File: modules/cloudbuild/main.tf [hunk 2]
```diff
@@ -277,9 +282,6 @@ resource "google_organization_iam_member" "cloudbuild_serviceusage_consumer" {
   org_id = var.org_id
   role   = "roles/serviceusage.serviceUsageConsumer"
   member = "serviceAccount:${module.cloudbuild_project.project_number}@cloudbuild.gserviceaccount.com"
-  depends_on = [
-    google_project_service.cloudbuild_apis,
```
Comment (rjerrems) [inline modules/cloudbuild/main.tf:281]: is it safe to remove these? Previously there was race conditions with APIs being enabled & the SA existing in time for permissions to be applied.

**Review by rjerrems — APPROVED** [review]
LGTM - A couple of small things but not blockers.

File: modules/cloudbuild/main.tf (hunk 1 above)
Comment (bharathkkb) [reply modules/cloudbuild/main.tf:267]: writer has reader permissions + update/create
https://cloud.google.com/artifact-registry/docs/access-control#roles

File: modules/cloudbuild/main.tf (hunk 2 above)
Comment (bharathkkb) [reply modules/cloudbuild/main.tf:281]: I didn't run issues  when testing locally. Within PF we already have a dependency on API enablement before returning `project_number` [here](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/4275a8d52d5455cb16f8af5f179ea882a4eaeedb/modules/core_project_factory/outputs.tf#L36). Best approach would be to use the service identity returned [directly](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/4275a8d52d5455cb16f8af5f179ea882a4eaeedb/modules/project_services/outputs.tf#L27). I will open a PR in PF to plumb that through.

**Comment by rjerrems** [comment]
Should we mark this as a breaking change @bharathkkb or I suppose the older image in GCR will remain regardless since it is built with gcloud?

Can you also rebase from master :)

File: modules/cloudbuild/main.tf (hunk 2 above)
Comment (bharathkkb) [reply modules/cloudbuild/main.tf:281]: new PF release will have that output
https://github.com/terraform-google-modules/terraform-google-project-factory/pull/549

**Comment by bharathkkb** [comment]
@rjerrems this should not be a breaking change as the older image will continue to live in GCR. Rerun would just create a new image in GAR and users can switch to GAR image if desired.

**Pull request merged**